clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpConnection.java

```diff
@@ -238,11 +240,13 @@ protected static Map<String, String> createDefaultHeaders(ClickHouseConfig confi
             // TODO check if auth-scheme is available and supported
             map.put("authorization", credentials.getAccessToken());
         } else if (!hasAuthorizationHeader) {
-            map.put("x-clickhouse-user", credentials.getUserName());
             if (config.isSsl() && !ClickHouseChecker.isNullOrEmpty(config.getSslCert())) {
+                map.put("x-clickhouse-user", credentials.getUserName());
                 map.put("x-clickhouse-ssl-certificate-auth", "on");
-            } else if (!ClickHouseChecker.isNullOrEmpty(credentials.getPassword())) {
-                map.put("x-clickhouse-key", credentials.getPassword());
+            } else {
+                String password = credentials.getPassword() == null ? "" : credentials.getPassword();
```

Comment:
Just to understand you have changed the logic to use Basic authorization to avoid the illegal characters.

Reply:
Yes, I've changed this logic to avoid problem with different characters. 
So even sequence `=?` is getting problem.
Additionally it does things more standard. 
Javascript Client uses this approach as default - https://github.com/ClickHouse/clickhouse-js/blob/main/packages/client-node/src/connection/node_base_connection.ts#L87